src/coreclr/src/System.Private.CoreLib/src/System/Type.CoreCLR.cs

```diff
@@ -20,27 +21,31 @@ public bool IsInterface
             }
         }
 
+        [RequiresUnreferencedCode("The type might be removed")]
         [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
         public static Type? GetType(string typeName, bool throwOnError, bool ignoreCase)
```

Comment:
This one is tough - I do agree that the example in JSON is one where we don't need to preserve those types. That said it's hard to say if the `throw: false` is a good enough indicator of the intent in this case. I can easily imagine that people use it just to suppress exceptions in favor of custom error handling, but the app still requires the type to exist for it to work.

Reply:
Yup, it would be an observable difference between trimmed and untrimmed without triggering a warning - we have been avoiding those.

If there's potential for behavioral differences, there needs to be a warning.

Reply:
It kind of sucks, but the only solution I can think of now is to use one of the "dangerous" APIs - like `Assembly.GetType` or maybe even better (more future proof) `Assembly.GetTypes` and effectively write the "GetType" functionality manually. Perf-wise this should not matter that much - especially if the result is cached (the above doesn't cache it, so it's pretty inefficient). Would obviously require suppress warning attributes as well.

I think the only "correct" solution is adding a public API which describes the intent - so something like `Type.GetTypeIfExists` (probably needs better name) which would intentionally be marked as 'trim friendly" without any intrinsic handling in the linker.

Reply:
We can already write a local GetTypeIfExists API 😉:

```csharp
[UnconditionalSuppressMessage(...)]
static Type GetTypeIfExists(string name) => Type.GetType(name, false);
```

Reply:
That's a good point - it's a little bit fragile as in theory we may improve linker to recognize this one day, but that's unlikely.

Reply:
Yes, we could add a trimming test to S.T.Json to make sure S.T.Json won't drag in those types into the app. The test would basically make sure the code path that could drag in Queue is active and then try to use a `GetTypeIfExists` API from the test to check if the types are there.

The test would start failing if we start dragging in the esoteric types and a new workaround will be needed at that point.